pybop/costs/fitting_costs.py

```diff
@@ -39,10 +39,23 @@ def _evaluate(self, x, grad=None):
         """
         prediction = self.problem.evaluate(x)
 
-        if len(prediction) < len(self._target):
-            return np.float64(np.inf)  # simulation stopped early
+        for key in prediction:
+            if key not in ["Time [s]", "Discharge capacity [A.h]"]:
```

Comment:
Filtering at this point is to avoid looping across the whole prediction dictionary, on second viewing it might be better to filter based on the `additional_variable` object.

Although, it might be worth looping across the whole dictionary in some cases specifically if ithe problem is standalone and hasn't been verified.

Reply:
The relevant variables here are those in the signal. For functioning models, it should only be necessary to check the length of one output (as all outputs should have the same length), so I suggest:

```
if len(prediction.get(signal[1], [])) < len(self._target.get(signal[1], [])):
     return np.float64(np.inf)  # simulation stopped early
```
It would also be ok to loop over all the signals.